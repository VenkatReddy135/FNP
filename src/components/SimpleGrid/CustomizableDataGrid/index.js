/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-param-reassign */
import React, { useEffect, useState, cloneElement } from "react";
import PropTypes from "prop-types";
import { Datagrid, useTranslate, useListContext } from "react-admin";
import { makeStyles } from "@material-ui/core/styles";
import isEmpty from "lodash/isEmpty";
import filter from "lodash/filter";
import get from "lodash/get";
import ColumnIcon from "@material-ui/icons/ViewColumn";
import Button from "@material-ui/core/Button";
import SelectionDialog from "./SelectionDialog";
import LocalStorage from "../../../utils/localstorage";
import { COCKPIT_SELECT_API } from "../../../config/GlobalConfig";

/**
 * makeStyles hook of material-ui to apply style for Reset form
 *
 *  @function useStyles
 */
const useStyles = makeStyles({
  containerWrapper: {
    float: "right",
    marginRight: "1rem",
    marginBottom: "1rem",
    marginTop: "-3rem",
    display: "flex",
    alignItems: "center",
    flexDirection: "row-reverse",
  },
});

/**
 * Array to selecttion function
 *
 * @param {Array} values -values to iterate
 * @returns {object} -return an object with key and value pair
 */
const arrayToSelection = (values) =>
  values.reduce((selectionAcc, columnName) => {
    selectionAcc[columnName] = true;
    return selectionAcc;
  }, {});

/**
 * Extract column name function
 *
 * @param {object} propsData props
 * @returns {object} column name
 */
const getColumnNames = (propsData) => {
  const { children } = propsData;
  return filter(React.Children.map(children, (field) => get(field, ["props", "source"])));
};

// CustomizableDatagrid allows to show/hide columns dynamically
// the preferences are stored in local storage

let resourceKey = null;
/**
 * Data Grid function
 *
 * @param {*} props -values to iterate
 * @returns {React.ReactElement} Data Grid with column name.
 */
const CustomizableDatagrid = (props) => {
  const translate = useTranslate();

  /**
   * Array to selecttion function
   *
   * @returns {object} Object with selected field.
   */
  const getInitialSelection = () => {
    const { defaultColumns, resource, storage, location } = props;
    if (resource.indexOf(window.REACT_APP_COCKPIT_SERVICE) !== -1 && resource.indexOf(COCKPIT_SELECT_API) !== -1) {
      resourceKey = resource + (location?.search || "");
    } else {
      resourceKey = resource;
    }

    const previousSelection = storage.get(resourceKey);

    // if we have a previously stored value, let's return it
    if (!isEmpty(previousSelection)) {
      return previousSelection;
    }

    // if defaultColumns are set let's return them
    if (!isEmpty(defaultColumns)) {
      return arrayToSelection(defaultColumns);
    }

    // otherwise we fallback on the default behaviour : display all columns
    return arrayToSelection(getColumnNames(props));
  };
  const [modalOpened, setModalOpened] = useState(false);
  const [selection, setSelection] = useState(getInitialSelection());
  const { resource, displayedFilters, filterValues, showFilter } = useListContext();
  /**
   * Array to selection function
   *
   * @returns {object} Object with selected field.
   */
  const getColumnLabels = () => {
    const { children } = props;
    return filter(
      React.Children.map(
        children,
        (field) =>
          field && {
            source: get(field, ["props", "source"]),
            label: get(field, ["props", "label"]),
            type: get(field, ["props", "type"]),
          },
      ),
      (item) => item && item.source,
    );
  };
  // updates the storage with the internal state value
  /**
   * Update data in localstore funciton
   *
   * @returns {undefined} returning undefined
   */
  const updateStorage = () => {
    props.storage.set(resourceKey, selection);
  };
  useEffect(() => {
    updateStorage();
    /* eslint-disable react-hooks/exhaustive-deps */
  }, [selection]);
  /**
   * Function to toggleColumns based on selected value
   *
   * @param {string} columnName column name
   */
  const toggleColumn = (columnName) => {
    const previousSelection = selection;
    const selectionTemp = {
      ...previousSelection,
      [columnName]: !previousSelection[columnName],
    };
    setSelection(selectionTemp);
  };
  /**
   * Function to update mdoal state
   *
   * @returns {Function} returning undefined
   */
  const handleOpen = () => setModalOpened(true);
  /**
   * Function to update modal state
   *
   * @returns {undefined} returning undefined
   */
  const handleClose = () => setModalOpened(false);
  const { children, rowStyleFunc, defaultColumns, additionalLink, filters, ...rest } = props;
  /**
   * Function to render the children
   *
   * @param {React.ReactElement} child react component
   * @returns {React.ReactElement} returning react element
   */
  const renderChild = (child) => {
    const source = get(child, ["props", "source"]);
    // Show children without source, or children explicitly visible
    if (!source || selection[source]) {
      return React.cloneElement(child, {});
    }

    return null;
  };

  const classes = useStyles();

  return (
    <div>
      <div className={classes.containerWrapper}>
        <Button
          title={translate("select_configuration_columns")}
          variant="outlined"
          mini
          aria-label="add"
          onClick={handleOpen}
        >
          <ColumnIcon />
        </Button>
        {filters &&
          cloneElement(filters, {
            resource,
            showFilter,
            displayedFilters,
            filterValues,
            context: "button",
          })}
        {additionalLink}
      </div>
      {modalOpened && (
        <SelectionDialog
          selection={selection}
          columns={getColumnLabels()}
          onColumnClicked={toggleColumn}
          onClose={handleClose}
        />
      )}
      <Datagrid {...rest} rowStyle={rowStyleFunc}>
        {React.Children.map(children, renderChild)}
      </Datagrid>
    </div>
  );
};

CustomizableDatagrid.propTypes = {
  defaultColumns: PropTypes.arrayOf(PropTypes.string),
  storage: PropTypes.shape({
    get: PropTypes.func.isRequired,
    set: PropTypes.func.isRequired,
  }),
  resource: PropTypes.string,
  children: PropTypes.arrayOf(PropTypes.object),
  rowStyleFunc: PropTypes.func,
  location: PropTypes.shape({
    hash: PropTypes.string,
    pathname: PropTypes.string,
    search: PropTypes.string,
    state: PropTypes.string || PropTypes.objectOf(PropTypes.any),
  }),
  additionalLink: PropTypes.element,
  filters: PropTypes.objectOf(PropTypes.any),
};

CustomizableDatagrid.defaultProps = {
  defaultColumns: [],
  storage: LocalStorage,
  resource: "",
  children: [],
  rowStyleFunc: () => {},
  location: {},
  additionalLink: <></>,
  filters: null,
};

export default CustomizableDatagrid;
