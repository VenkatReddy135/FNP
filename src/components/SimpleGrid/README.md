# Simple Grid

Simple Grid component renders a react-admin List component with Customizable Datagrid. The List component has sortable columns and pagination that react-admin offers with the List.

## Props

Simple Grid needs the following props to render the grid -

1. configurationForGrid :

- This is an array of configuration object for the columns you want to render on your Grid. Each column object should contain at least the source, type and the label. In the 'source', you will provide the field name that you get from the API. The 'type' would contain the Type of component you want to render in that column and 'label' contains the title of the column header. For e.g.,  
   {
  source: "name",
  type: "TextField",
  label: "Name of the user"
  }
- If the type of the column is a custom component and it requires some custom props specific to that component, you can pass the same in the configuration object for that column.
- You can also add an onClickHandler in the configuration object, if you wish to add any click events for that particular column. For e.g.,
  {
  source: "website",
  type: "UrlField",
  onClick: onClickHandler
  }
- We loop on this 'configurationForGrid' array to render the grid columns. We have a 'componentsMap' in the simple grid component which contains the list of imported components you want to render in grid columns. Make sure to add the type of component you want to render in this 'componentsMap'.

2. actionButtonsForGrid :

- This is an array of configuration object for the action buttons you want to render with your Grid. The object must contain a 'type' prop to indicate the type of button component you want to render.
- You can add additional props in the object depending on your requirements. For e.g.,
  {
  type: "CreateButton",
  label: translate("new_category"),
  icon: <></>,
  className: classes.buttonStyle,
  },
- You can also add onClickHandlers in the object if you want to add a onClick event with the button.
  For e.g.,
  {
  type: "ImportButton",
  configurationForExpandMenu,
  onClick: onClickHandler,
  },
- If the type of the button is a custom component and it requires some custom props specific to that component, you can pass the same in the configuration object for that button.
- 'actionButtonsForGrid' array then gets passed to the GridActions component from the simple grid where we loop on this configuration array and render the button components. We have a 'componentsMap' in the GridActions component which contains the list of imported Button components. Make sure to add the type of button you want to render in this 'componentsMap'. For e.g.,
  const componentsMap = {
  CreateButton,
  ImportButton,
  Button,
  };

3. gridTitle :

- This contains the name you want to add in the Grid Title
- It is a string type field

4. filters :

- This prop is an optional prop which accepts a component that returns <Filter /> component need by the SimpleGrid list 'filters' prop.
- By default, this prop value is 'null'.
- If this prop is being passed from the list, it will be shown as Grid filters, otherwise the default <GridFilters/> component will display the filters (contains only a SearchInput).
- These filter values are further passed to <GridActions/> component to display filters using 'Add filter' option button.

5. showAddFilterButton:

- This flag determines whether you want to show the 'Add filter' button on the Simple Grid or not.
- If showAddFilterButton = true, 'Add filter' button will be shown.
  If showAddFilterButton = false (by default it is false), 'Add filter' button will not be shown.
- So if you want to show the 'Add filter' button, pass this prop 'showAddFilterButton' as true to <SimpleGrid /> from your list page component.
- We pass this 'filters' prop to <GridActions /> component based on this 'showAddFilterButton' prop value as shown below :
  <GridActions actionButtonsForGrid={actionButtonsForGrid} filters={showAddFilterButton ? filters : null} />
- If 'showAddFilterButton' is true, only then the Grid actions will show the 'Add filter' button and hence the filter component passed from list.

6. actionButtonsForEmptyGrid :

- This contains configuration array which is used for adding action buttons on the grid
- e.g., const actionButtonsForEmptyGrid = [
  {
  type: "CreateButton",
  label: translate("redirect_campaign.create_title"),
  icon: <></>,
  onClick: createHandler,
  variant: "outlined",
  },
  ];

7. searchLabel :

- This contains the string value which is passed to the gridFilter component as a label.
- e.g., searchLabel = translate("category_search_label");

8. resource :

- This contains the resource name for which the list will get rendered.
- e.g., resource={`${window.REACT_APP_GALLERIA_SERVICE}/categories`}

9. bulkActionButtons :

- This is a boolean prop which is used to render action buttons conditionally
- bulkActionButtons: null,

10. sortField :

- This defines the sort field and its order
- sortField: { field: "id", order: "ASC" },

11. filters :

- This prop contains custom filter component value which user can pass to the grid
- e.g., filters={<DateAndSearchFilters searchLabel={viewHistorySearchLabel} {...props} />}

12. getFilter :

- This is handler function for the filter to fect filter value
- e.g., getFilter: () => {},

13. additionalLink :

- To add advancesearch we use this link
- This is optional
- e.g : <Link className={classesGlobal.redirectLink} to="/beautyplus/v1/advancesearch">
  <span className={classesGlobal.search}>Advanced Search</span>
    </Link>

14. limit :

- An element that is displayed if there is no data to show (default: <PaginationLimit>)
- This is optional
- e.g : pagination={<CustomPagination rowsPerPageOptions={paginationLimitValue} limit={limit} />}

## Customizable DataGrid

- We have used a third party component 'CustomizableDatagrid' from 'ra-customizable-datagrid' package. - - This component enables the hide/show functionality for the grid columns.
- This adds a ColumnIcon at the right top corner of the grid on click of which opens a dialog. This dialog contains the list of all the column names with a checkbox. The columns that are checked will be displayed in the grid & the unchecked columns will be hidden from the grid.
