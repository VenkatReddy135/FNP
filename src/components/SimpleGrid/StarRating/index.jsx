import React from "react";
import Rating from "@material-ui/lab/Rating";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles(() => ({
  ratingContainer: {
    marginTop: "20px",
  },
}));

/**
 * Function to convert rating values into Star rating to display in Grid
 *
 * @name StarRating
 * @param {object} props - Passing record, source
 * @returns {React.ReactElement} returns star rating component
 */
const StarRating = (props) => {
  const { record, source } = props;
  const classes = useStyles();

  return (
    <Rating className={classes.ratingContainer} data-testid="star-rating" defaultValue={record[source]} readOnly />
  );
};

StarRating.propTypes = {
  record: PropTypes.objectOf(PropTypes.any),
  source: PropTypes.string.isRequired,
};
StarRating.defaultProps = {
  record: {},
};

export default StarRating;
