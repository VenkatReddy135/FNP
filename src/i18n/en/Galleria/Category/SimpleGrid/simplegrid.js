const simpleGridMessages = {
  new_category: "New Category",
  new_relationship: "New Relation",
  category_search_label: "Search by Category ID, Category Name or Category URL",
  relation_search_label: "Search by Category Id, Name and Type",
  category_management: "Category Management",
  category_relationship_management: "Category Relationship Management",
  category_id: "Category ID",
  category_url: "Category URL",
  category_name: "Category Name",
  relation_category_name: "Category Name",
  category_type: "Category Type",
  relation_type: "Relation Type",
  primary: "Primary",
  sequence: "Sequence",
  is_enabled: "Is Enabled",
  created_by: "Created By",
  created_date: "Created Date",
  parent_category_name: "Parent Category Name",
  workflow_status: "Workflow Status",
  enabled: "Enabled",
  import: "Import",
  export: "Export",
  choose_file_btn: "Choose File",
  import_req_title: "Import request accepted",
  import_success_message: "Import Request submitted successfully. You will be notified when the job is processed.",
  choose_file: "Choose file for import",
  error_message: "An error occurred. Please check after sometime",
  lorem_ipsum_capitalize: "Lorem Ipsum",
  lorem_ipsum_lowercase: "lorem Ipsum",
  content_management: "Content Management",
  add_associate_content: "Add/Associate Content",
  content_type: "Content Type",
  content_details: "Content Details",
  content_id: "Content Id",
  category_content_name: "Category Content Name",
  dialog_title: "Choose file for import",
  language: "Language",
  from_date: "From Date",
  btn_cancel: "Cancel",
  btn_ok: "OK",
  content_search_label: "Search by Associated Content Id, Content Id and Category Content Name",
  url_redirect_tool: "URL Redirect Tool",
  new_url_redirect: "NEW URL REDIRECT",
  source_url: "Source URL",
  target_url: "Target URL",
  entity_type: "Entity Type",
  redirect_type: "Redirect Type",
  comments: "Comments",
  last_modified_by: "Last Modified By",
  url_redirect_search_label: "Search by created by and last modified by",
  configuration: "Configuration",
  no_results_found: "No results found",
  association: "association",
  btn_export_req: "Export request accepted ",
  upload_title: "File upload in progress",
};

export default simpleGridMessages;
