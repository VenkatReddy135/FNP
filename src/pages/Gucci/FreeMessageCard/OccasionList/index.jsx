/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react-hooks/exhaustive-deps */
import React, { useState } from "react";
import { useTranslate, useMutation, useRefresh, useNotify } from "react-admin";
import { useDispatch } from "react-redux";
import SimpleGrid from "../../../../components/SimpleGrid";
import SimpleModel from "../../../../components/CreateModal";
import { setOccasionData } from "../../../../actions/freemessagecard";
import CommonDialogContent from "../../../../components/CommonDialogContent";
import { onSuccess, onFailure } from "../../../../utils/CustomHooks";
import formatDateValue from "../../../../utils/formatDateTime";

/**
 * Component for Occasions List contains a simple grid with configurations for Occasions
 *
 * @param {object} props all the props needed for Occasions List
 * @returns {React.ReactElement} returns a Occasions List component
 */
const OccasionList = (props) => {
  const translate = useTranslate();
  const notify = useNotify();
  const [mutate] = useMutation();
  const refresh = useRefresh();
  const [isOpen, setIsOpen] = useState(false);
  const [confirmDialogObj, setConfirmDialog] = useState({});
  const [selectedRecord, setSelectedRecord] = useState(null);
  const configurationForKebabMenu = [];
  const dispatch = useDispatch();

  /**
   * To change the value for status enabled/disable
   *
   * @param {boolean} _event enable/disable value for selected record
   * @param {object} record selected record object
   * @function handleStatusChange
   */
  const handleStatusChange = (_event, record) => {
    const dialogObject = {
      dialogContent: (
        <CommonDialogContent
          message={`${translate("update_message")} ${translate("configuration")}${translate("question_mark")}`}
        />
      ),
      showButtons: true,
      closeText: translate("cancel"),
      actionText: translate("update"),
    };
    setConfirmDialog(dialogObject);
    setSelectedRecord(record);
    setIsOpen(true);
  };

  /**
   * To close confirm dialog.
   *
   * @function closeDialogHandler
   */
  const closeDialogHandler = () => {
    setIsOpen(false);
  };

  /**
   * To send record object across message listing and other pages.
   *
   * @function clickHandler
   * @param {object} _event onClick event object
   * @param {object} record object containing selected occasion
   */
  const clickHandler = (_event, record) => {
    dispatch(setOccasionData({ selectedOccasion: record }));
  };

  /**
   *
   * @function handleSuccess to handle success of the API
   * @param {object} res response object containing status and response data
   */
  const handleSuccess = (res) => {
    if (res.status === "success" && res.data) refresh();
  };

  /**
   * To update status for a selected record
   *
   * @function updateHandler
   */
  const updateHandler = () => {
    mutate(
      {
        type: "put",
        resource: `${window.REACT_APP_BEAUTYPLUS_SERVICE}/sites/messages/occasion/${selectedRecord.id}`,
        payload: {},
      },
      {
        onSuccess: (response) => {
          onSuccess({ response, notify, translate, handleSuccess });
        },
        onFailure: (error) => {
          onFailure({ error, notify, translate });
        },
      },
    );
    setIsOpen(false);
  };

  /**
   *Function to handle rendering for function field.
   *
   * @function customRenderHandler
   * @param {object} record individual record from list
   * @returns {Function} formatDateValue returns formatted date/time value
   */
  const customRenderHandler = (record) => {
    return record.updatedAt ? `${formatDateValue(record.updatedAt)}` : `${formatDateValue(record.createdAt)}`;
  };

  const configurationForOccasionGrid = [
    { source: "occasionName", type: "TextField", label: translate("category_occ") },
    {
      source: "enabled",
      type: "SwitchComp",
      compareKey: "enabled",
      label: translate("status"),
      disable: false,
      onChange: handleStatusChange,
    },
    {
      source: "occasionId",
      type: "KebabMenuWithLink",
      label: translate("messages"),
      tabPath: "/view",
      isLink: true,
      configurationForKebabMenu,
      isAnchorLink: true,
      displayText: translate("Manage"),
      onLinkClick: clickHandler,
    },
    { source: "createdBy", type: "TextField", label: translate("created_by") },
    { source: "createdAt", type: "CustomDateField", label: translate("created_date") },
    { source: "updatedBy", type: "TextField", label: translate("modified_by") },
    {
      source: "updatedAt",
      type: "FunctionField",
      label: translate("last_modified_date"),
      sortBy: "updatedAt",
      render: customRenderHandler,
    },
  ];

  const occasionGridTitle = translate("free_message_config");
  const occasionSearchLabel = translate("search");
  // dispatch(setOccasionData({ selectedOccasion: {} }));

  return (
    <>
      <SimpleGrid
        {...props}
        resource={`${window.REACT_APP_BEAUTYPLUS_SERVICE}/sites/messages`}
        configurationForGrid={configurationForOccasionGrid}
        actionButtonsForGrid={[]}
        actionButtonsForEmptyGrid={[]}
        gridTitle={occasionGridTitle}
        searchLabel={occasionSearchLabel}
        sortField={{ field: "occasionName", order: "ASC" }}
        isSmallerSearch
      />
      <SimpleModel
        {...confirmDialogObj}
        openModal={isOpen}
        handleClose={closeDialogHandler}
        handleAction={updateHandler}
      />
    </>
  );
};

export default React.memo(OccasionList);
