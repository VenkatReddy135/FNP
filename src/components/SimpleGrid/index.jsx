/* eslint-disable react/jsx-props-no-spreading */

import PropTypes from "prop-types";
import React from "react";
import {
  List,
  DateField,
  BooleanField,
  NumberField,
  BooleanInput,
  ChipField,
  UrlField,
  SimpleForm,
  TextInput,
  EmailField,
  ResourceContextProvider,
  Pagination,
  PaginationActions,
  FunctionField,
  useTranslate,
} from "react-admin";
import { Typography, Divider, Grid } from "@material-ui/core";
import CustomizableDatagrid from "./CustomizableDataGrid";
import GridActions from "../GridActionButtons";
import KebabMenuWithLink from "../KebabMenuWithLink";
import TextField from "../TextField";
import GridFilters from "../GridFilters";
import useStyles from "../../assets/theme/common";
import SwitchComp from "../switch";
import Status from "../WorkFlowStatus";
import TooltipTextField from "../TooltipTextField";
import CustomSourceData from "./SourceField/CustomSourceData";
import CustomDateField from "../DateFieldConversion";
import UrlComponent from "../UrlComponent";
import TextWithJson from "../TextWithJson";
import EmptyPage from "../EmptyPage";
import StatusField from "../StatusField";
import StarRating from "./StarRating";
import ActionButton from "../../pages/Moody/ActionButton";
import LinkButton from "../LinkButton";
import ExportDropdown from "../ExportDropdown";
import CustomGridInputField from "../CustomGridInputField";
import { defaultRowPerPageOption } from "../../config/GlobalConfig";

const componentsMap = {
  ActionButton,
  TextField,
  DateField,
  BooleanField,
  BooleanInput,
  NumberField,
  ChipField,
  UrlField,
  SimpleForm,
  TextInput,
  EmailField,
  KebabMenuWithLink,
  SwitchComp,
  CustomDateField,
  CustomSourceData,
  Status,
  TooltipTextField,
  StatusField,
  CustomGridInputField,
  UrlComponent,
  TextWithJson,
  StarRating,
  LinkButton,
  ExportDropdown,
  FunctionField,
};

/**
 * Component for Custom Pagination Actions
 *
 * @param {object} props all the props required by Simple Grid
 * @returns {React.ReactElement} returns a React component for Custom Pagination
 */
const CustomPaginationActions = (props) => <PaginationActions {...props} color="secondary" />;

/**
 * Component for Custom Pagination
 *
 * @param {object} props all the props required by Simple Grid
 * @returns {React.ReactElement} returns a React component for Simple Grid
 */
const CustomPagination = (props) => <Pagination {...props} ActionsComponent={CustomPaginationActions} />;

/**
 * Component for Grid Filters contains list of all the filters you want to see in the grid
 *
 * @param {*} props all the props required by Simple Grid
 * @returns {React.ReactElement} returns a React component for Simple Grid
 */
const SimpleGrid = ({
  configurationForGrid,
  actionButtonsForGrid,
  actionButtonsForEmptyGrid,
  gridTitle,
  searchLabel,
  bulkActionButtons,
  sortField,
  resource,
  basePath,
  filters,
  getFilter,
  showAddFilterButton,
  isSearchEnabled,
  clearSearchFilter,
  resetClearSearchFilter,
  rowStyleFunc,
  additionalLink,
  rowsPerPageOptions,
  limit,
  ...props
}) => {
  const classes = useStyles();
  const translate = useTranslate();
  const NO_RESULTS_FOUND_MESSAGE = translate("no_results_found");

  /**
   * @param {string} value param required for getFilterVal function
   * @function getFilterVal function fetches the filterValues for the search filter
   */
  const getFilterVal = (value) => {
    getFilter(value);
  };

  const paginationLimitValue = rowsPerPageOptions.length > 0 ? rowsPerPageOptions : defaultRowPerPageOption;
  const perpageDefault = rowsPerPageOptions.length > 0 ? rowsPerPageOptions[0] : defaultRowPerPageOption[1];

  return (
    <>
      <Grid item className={classes.gridStyle}>
        <Typography variant="h5" color="inherit" className={classes.titleLineHeight}>
          {gridTitle}
        </Typography>
      </Grid>
      {gridTitle && <Divider variant="fullWidth" />}
      <ResourceContextProvider resource={resource}>
        <List
          {...props}
          resource={resource}
          basePath={basePath}
          sort={sortField}
          bulkActionButtons={bulkActionButtons}
          pagination={
            <CustomPagination rowsPerPageOptions={paginationLimitValue} limit={limit || NO_RESULTS_FOUND_MESSAGE} />
          }
          filters={
            filters || (
              <GridFilters
                isSearchEnabled={isSearchEnabled}
                clearSearchFilter={clearSearchFilter}
                resetClearSearchFilter={resetClearSearchFilter}
                getFilterVal={getFilterVal}
                searchLabel={searchLabel}
                {...props}
              />
            )
          }
          actions={<GridActions actionButtonsForGrid={actionButtonsForGrid} />}
          component="div"
          empty={<EmptyPage actionButtonsForEmptyGrid={actionButtonsForEmptyGrid} additionalLink={additionalLink} />}
          perPage={perpageDefault}
        >
          <CustomizableDatagrid
            {...props}
            className={classes.grid}
            filters={showAddFilterButton ? filters : null}
            additionalLink={additionalLink}
            rowStyleFunc={rowStyleFunc}
          >
            {configurationForGrid.map((field) => {
              const componentType = field.type;
              const Component = componentsMap[componentType];
              return <Component key={field} {...field} />;
            })}
          </CustomizableDatagrid>
        </List>
      </ResourceContextProvider>
    </>
  );
};

SimpleGrid.propTypes = {
  configurationForGrid: PropTypes.arrayOf(PropTypes.object).isRequired,
  actionButtonsForGrid: PropTypes.arrayOf(PropTypes.object).isRequired,
  actionButtonsForEmptyGrid: PropTypes.arrayOf(PropTypes.object),
  gridTitle: PropTypes.string.isRequired,
  searchLabel: PropTypes.string,
  resource: PropTypes.string.isRequired,
  basePath: PropTypes.string,
  bulkActionButtons: PropTypes.oneOfType([PropTypes.element, PropTypes.bool]),
  sortField: PropTypes.objectOf(PropTypes.string),
  filters: PropTypes.objectOf(PropTypes.any),
  getFilter: PropTypes.func,
  showAddFilterButton: PropTypes.bool,
  isSearchEnabled: PropTypes.bool,
  clearSearchFilter: PropTypes.bool,
  resetClearSearchFilter: PropTypes.func,
  rowStyleFunc: PropTypes.func,
  additionalLink: PropTypes.element,
  rowsPerPageOptions: PropTypes.arrayOf(PropTypes.number),
  limit: PropTypes.string,
};
SimpleGrid.defaultProps = {
  bulkActionButtons: false,
  searchLabel: "",
  showAddFilterButton: false,
  sortField: { field: "id", order: "DESC" },
  filters: null,
  basePath: "",
  actionButtonsForEmptyGrid: null,
  getFilter: () => {},
  isSearchEnabled: true,
  clearSearchFilter: false,
  resetClearSearchFilter: () => {},
  rowStyleFunc: () => {},
  additionalLink: <></>,
  rowsPerPageOptions: [],
  limit: "",
};

export default SimpleGrid;
