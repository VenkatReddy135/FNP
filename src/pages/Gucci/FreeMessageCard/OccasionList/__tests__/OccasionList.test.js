/**
 * @jest-environment jsdom
 */
import React from "react";

import SimpleGrid from "../../../../../components/SimpleGrid";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import OccasionList from "..";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

jest.mock("react-admin", () => ({
  useNotify: jest.fn(),
  useTranslate: () => (label) => label,
  useMutation: () => [() => {}],
  useRefresh: () => [() => {}],
  List: () => <></>,
  ResourceContextProvider: () => <></>,
  Component: () => <></>,
}));

describe("Occasion List test cases", () => {
  const initialState = {
    occasionData: {
      selectionOccasion: {
        occasionId: "100th-birthday",
        occasionName: "100th Birthday",
      },
    },
  };

  const mockStore = configureStore();
  const store = mockStore(initialState);
  let component, testInstance;

  beforeEach(() => {
    component = renderer.create(
      <Provider store={store}>
        <OccasionList />
      </Provider>,
    );
    testInstance = component.root;
  });
  afterEach(() => {
    cleanup();
  });
  it("should render OccasionList component", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should contain grid title", () => {
    expect(testInstance.findByType(SimpleGrid).props.gridTitle).toBe("free_message_config");
  });
  it("should contain search label", () => {
    expect(testInstance.findByType(SimpleGrid).props.searchLabel).toBe("search");
  });
  it("should not have any buttons on filled grid", () => {
    expect(testInstance.findByType(SimpleGrid).props.actionButtonsForGrid).toEqual([]);
  });
  it("should have smaller search field", () => {
    expect(testInstance.findByType(SimpleGrid).props.isSmallerSearch).toEqual(true);
  });
  it("should sort based on occasion name column by default", () => {
    expect(testInstance.findByType(SimpleGrid).props.sortField.field).toBe("occasionName");
  });
  it("should contain 7 columns", () => {
    expect(testInstance.findByType(SimpleGrid).props.configurationForGrid.length).toEqual(7);
  });
  it("should have first column header as Occasion", () => {
    expect(testInstance.findByType(SimpleGrid).props.configurationForGrid[0].label).toBe("category_occ");
  });
  it("should have second column header as Status", () => {
    expect(testInstance.findByType(SimpleGrid).props.configurationForGrid[1].label).toBe("status");
  });
  it("should have third column header as Messages", () => {
    expect(testInstance.findByType(SimpleGrid).props.configurationForGrid[2].label).toBe("messages");
  });
  it("should have link as values for Messages column", () => {
    expect(testInstance.findByType(SimpleGrid).props.configurationForGrid[2].isAnchorLink).toBe(true);
  });
});
