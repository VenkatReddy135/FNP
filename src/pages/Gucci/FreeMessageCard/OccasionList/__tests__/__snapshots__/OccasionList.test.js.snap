// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Occasion List test cases should render OccasionList component 1`] = `
Array [
  <div
    className="MuiGrid-root makeStyles-gridStyle-32 MuiGrid-item"
  >
    <h5
      className="MuiTypography-root makeStyles-titleLineHeight-35 MuiTypography-h5 MuiTypography-colorInherit"
    >
      free_message_config
    </h5>
  </div>,
  <hr
    className="MuiDivider-root"
  />,
]
`;
